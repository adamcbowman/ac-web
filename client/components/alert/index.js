import React from 'react'
import PropTypes from 'prop-types'
import styles from './Alert.css'

// Types
const DANGER = 'DANGER'
const INFO = 'INFO'
const WARNING = 'WARNING'
const SUCCESS = 'SUCCESS'

export const Danger = alert(DANGER)
export const Info = alert(INFO)
export const Warning = alert(WARNING)
export const Success = alert(SUCCESS)

// Component
Alert.propTypes = {
    type: PropTypes.oneOf([DANGER, INFO, WARNING, SUCCESS]).isRequired,
    children: PropTypes.node.isRequired,
}

function Alert({ type, children, ...props }) {
    return (
        <div className={classNames.get(type)} {...props}>
            {children}
        </div>
    )
}

// HOC
function alert(type) {
    return props => <Alert {...props} type={type} />
}

// Styles
const classNames = new Map([
    [DANGER, styles.Danger],
    [INFO, styles.Info],
    [WARNING, styles.Warning],
    [SUCCESS, styles.Success],
])
