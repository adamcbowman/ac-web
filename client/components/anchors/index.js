export Mailto from './Mailto'
export Phone from './Phone'
