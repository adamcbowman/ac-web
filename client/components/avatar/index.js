import React, { useState } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames/bind'
import { initials } from 'utils/string'
import styles from './Avatar.css'

Avatar.propTypes = {
    name: PropTypes.string.isRequired,
    url: PropTypes.string,
    size: PropTypes.number,
}

export default function Avatar({ size = 60, url, name }) {
    const [on, set] = useState(Boolean(url))
    const style = {
        height: size,
        width: size,
        fontSize: size < 50 ? '0.75em' : '1em',
    }
    const className = classNames({
        Initials: on,
        Avatar: !on,
    })

    return (
        <div className={className} data-initials={initials(name)} style={style}>
            {url && (
                <img
                    src={url}
                    alt={initials(name)}
                    title={name}
                    onLoad={() => set(false)}
                    onError={() => set(false)}
                />
            )}
        </div>
    )
}

// Style
const classNames = classnames.bind(styles)
