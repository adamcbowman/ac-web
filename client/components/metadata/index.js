export Metadata from './Metadata'
export Entry from './Entry'
export ShareEntry from './ShareEntry'
export TimestampEntry from './TimestampEntry'
