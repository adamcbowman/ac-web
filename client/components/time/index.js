import React from 'react'
import startOfDay from 'date-fns/start_of_day'
import parse from 'date-fns/parse'
import { DATE, DATETIME, TIME } from 'utils/date'
import Base from './Time'

export Relative from './Relative'
export Range from './Range'

export function Time(props) {
    return <Base format={TIME} {...props} />
}

export function DateTime(props) {
    return <Base format={dateTimeFormatGetter} {...props} />
}

export function Day(props) {
    return <Base format="dddd" {...props} />
}

export function DateElement(props) {
    return <Base format={DATE} {...props} />
}

// Utils
export function dateTimeFormatGetter(date) {
    date = parse(date)

    if (startOfDay(date).getTime() === date.getTime()) {
        return DATE
    }

    return DATETIME
}
