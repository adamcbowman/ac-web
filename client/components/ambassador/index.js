import React from 'react'
import PropTypes from 'prop-types'
import { Image } from 'prismic/components/base'
import { Set, Item } from 'components/social'
import { FragmentIdentifier } from 'router'
import styles from './Ambassador.css'

Ambassador.propTypes = {
    fullName: PropTypes.string.isRequired,
    socials: PropTypes.arrayOf(PropTypes.string),
    children: PropTypes.node.isRequired,
    avatar: PropTypes.object.isRequired,
    banner: PropTypes.object.isRequired,
    hash: PropTypes.string,
}

export default function Ambassador({
    fullName,
    socials = [],
    banner,
    avatar,
    children,
}) {
    const hash = fullName.toLowerCase().replace(/\s/g, '-')
    function title(name) {
        return `Visit ${fullName} on ${name}`
    }

    return (
        <section className={styles.Ambassador}>
            <div className={styles.Biography}>
                <div className={styles.Avatar}>
                    {avatar && <Image {...avatar} />}
                    <Set>
                        {socials.map(link => (
                            <Item key={link} link={link} title={title} />
                        ))}
                    </Set>
                </div>
                <div className={styles.Content}>
                    <h2>
                        <FragmentIdentifier hash={hash}>
                            {fullName}
                        </FragmentIdentifier>
                    </h2>
                    {children}
                </div>
            </div>
            {banner && <Image className={styles.Banner} {...banner} />}
        </section>
    )
}
