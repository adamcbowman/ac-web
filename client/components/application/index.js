import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import * as Page from 'components/page'
import * as Text from 'components/text'
import * as Layouts from 'layouts/pages'
import Navbar from 'components/navbar'
import typography from 'components/text/Text.css'

// TODO Reuse existing page layouts
// TODO Move these to layouts
export function Fallback({ error, navbar, children }) {
    return (
        <Fragment>
            {navbar || <Navbar />}
            <Layouts.Error>
                <Page.Main>
                    <h1>Uh oh! We never thought that would happen...</h1>
                    <Page.Headline>
                        An error occured on the page you are visiting.
                        <br />
                        We have been notified about that error and we will try
                        to fix as soon as possible.
                    </Page.Headline>
                    <details>
                        <summary>More details</summary>
                        <Text.Error>{error.name}</Text.Error>
                        <Text.Error>{error.message}</Text.Error>
                    </details>
                    {children}
                </Page.Main>
            </Layouts.Error>
        </Fragment>
    )
}

ErrorDetails.propTypes = {
    error: PropTypes.instanceOf(Error),
    className: PropTypes.string,
    summary: PropTypes.string,
    children: PropTypes.node,
}

export function ErrorDetails({
    error,
    className,
    summary = 'An error occured.',
    children,
}) {
    return (
        <details className={classnames(typography.Error, className)}>
            <summary>{summary}</summary>
            {error && <p>{error.message}</p>}
            {children}
        </details>
    )
}
