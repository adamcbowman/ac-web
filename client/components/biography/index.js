import React, { memo } from 'react'
import PropTypes from 'prop-types'
import Avatar from 'components/avatar'
import { Mailto, Phone } from 'components/anchors'
import styles from './Biography.css'

// FIXME: This component is asking for too much! Find a way to be more generic. Look at Blocquote.

Biography.propTypes = {
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    email: PropTypes.string,
    title: PropTypes.string,
    avatar: PropTypes.string,
    workPhoneNumber: PropTypes.string,
    children: PropTypes.node,
}

function Biography({
    email,
    workPhoneNumber,
    children,
    firstName,
    lastName,
    title,
    avatar,
}) {
    const fullName = `${firstName} ${lastName}`

    return (
        <section className={styles.Container}>
            <div className={styles.Media}>
                <Avatar url={avatar} name={fullName} />
            </div>
            <div className={styles.Content}>
                <header className={styles.Header}>
                    <span className={styles.Name}>{fullName}</span>
                    {title && <span className={styles.Title}>{title}</span>}
                </header>
                <address className={styles.Address}>
                    {email && (
                        <Mailto
                            email={email}
                            title={`Send ${fullName} an email`}
                        />
                    )}
                    {workPhoneNumber && (
                        <Phone
                            phone={workPhoneNumber}
                            ext={null}
                            title={`Give ${fullName} a call`}
                        />
                    )}
                </address>
                {children}
            </div>
        </section>
    )
}

export default memo(Biography)
