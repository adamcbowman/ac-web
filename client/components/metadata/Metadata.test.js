import React from 'react'
import Renderer from 'react-test-renderer'
import { Metadata, Entry, ShareEntry } from './'
import { DateTime } from 'components/time'

test('metadata component', () => {
    const metadata = Renderer.create(
        <Metadata>
            <Entry term="Date issued">
                <DateTime value={new Date(2017, 1, 1)} />
            </Entry>
            <Entry term="Valid until">
                <DateTime value={new Date(2017, 1, 2)} />
            </Entry>
            <Entry term="Forecaster">James Floyer</Entry>
            <ShareEntry />
        </Metadata>
    )

    expect(metadata).toMatchSnapshot()
})
