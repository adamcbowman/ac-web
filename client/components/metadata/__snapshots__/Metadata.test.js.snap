// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metadata component 1`] = `
<div
  className="Metadata"
>
  <dl
    className="Entry"
  >
    <dt
      className="Term"
    >
      Date issued
    </dt>
    <dd
      className="Description"
    >
      <time
        dateTime="2017-02-01T00:00:00.000-08:00"
      >
        Wednesday, February 1, 2017
      </time>
    </dd>
  </dl>
  <dl
    className="Entry"
  >
    <dt
      className="Term"
    >
      Valid until
    </dt>
    <dd
      className="Description"
    >
      <time
        dateTime="2017-02-02T00:00:00.000-08:00"
      >
        Thursday, February 2, 2017
      </time>
    </dd>
  </dl>
  <dl
    className="Entry"
  >
    <dt
      className="Term"
    >
      Forecaster
    </dt>
    <dd
      className="Description"
    >
      James Floyer
    </dd>
  </dl>
  <dl
    className="Entry"
  >
    <dt
      className="Term"
    />
    <dd
      className="Description"
    >
      <div
        className="Set"
      >
        <a
          className="Item"
          href="https://www.facebook.com/sharer.php?u=http://localhost/"
          style={
            Object {
              "padding": "0 0.5em",
            }
          }
          target="_blank"
          title="Share on Facebook"
        >
          <svg
            className={undefined}
            fill="none"
            height={24}
            viewBox="0 0 266.893 266.895"
            width={24}
          >
            <path
              d="M248.082,262.307c7.854,0,14.223-6.369,14.223-14.225V18.812 c0-7.857-6.368-14.224-14.223-14.224H18.812c-7.857,0-14.224,6.367-14.224,14.224v229.27c0,7.855,6.366,14.225,14.224,14.225 H248.082z"
              fill="#3C5A99"
            />
            <path
              d="M182.409,262.307v-99.803h33.499l5.016-38.895h-38.515V98.777c0-11.261,3.127-18.935,19.275-18.935 l20.596-0.009V45.045c-3.562-0.474-15.788-1.533-30.012-1.533c-29.695,0-50.025,18.126-50.025,51.413v28.684h-33.585v38.895h33.585 v99.803H182.409z"
              fill="#FFFFFF"
            />
          </svg>
        </a>
        <a
          className="Item"
          href="https://twitter.com/intent/tweet?url=http://localhost/"
          style={
            Object {
              "padding": "0 0.5em",
            }
          }
          target="_blank"
          title="Share on Twitter"
        >
          <svg
            className={undefined}
            fill="#55ACEE"
            height={24}
            viewBox="0 0 335 276"
            width={24}
          >
            <path
              d="m302 70a195 195 0 0 1 -299 175 142 142 0 0 0 97 -30 70 70 0 0 1 -58 -47 70 70 0 0 0 31 -2 70 70 0 0 1 -57 -66 70 70 0 0 0 28 5 70 70 0 0 1 -18 -90 195 195 0 0 0 141 72 67 67 0 0 1 116 -62 117 117 0 0 0 43 -17 65 65 0 0 1 -31 38 117 117 0 0 0 39 -11 65 65 0 0 1 -32 35"
            />
          </svg>
        </a>
        <a
          className="Item"
          href="https://plus.google.com/share?url=http://localhost/"
          style={
            Object {
              "padding": "0 0.5em",
            }
          }
          target="_blank"
          title="Share on Google Plus"
        >
          <svg
            className={undefined}
            fill="#DC4E41"
            height={24}
            viewBox="0 0 55 55"
            width={24}
          >
            <g>
              <path
                d="M19.6671,25.7867c-0.0075,1.7935,0,3.5869,0.0076,5.3803c3.0067,0.098,6.0208,0.0527,9.0275,0.098   c-1.3262,6.6689-10.3989,8.8315-15.199,4.4761C8.5674,31.9206,8.801,23.5412,13.9327,19.992   c3.5869-2.8635,8.6884-2.1552,12.2752,0.324c1.4092-1.3036,2.7278-2.6977,4.0013-4.1445   c-2.984-2.3812-6.6462-4.0767-10.5421-3.8958c-8.1307-0.2713-15.6059,6.8497-15.7415,14.9805   c-0.52,6.6462,3.8506,13.1644,10.0222,15.5155c6.1489,2.3661,14.031,0.7535,17.957-4.77c2.5922-3.4889,3.1498-7.98,2.8484-12.1999   C29.7194,25.7641,24.6933,25.7716,19.6671,25.7867z"
              />
              <path
                d="M49.0704,25.7641c-0.0151-1.4996-0.0226-3.0067-0.0301-4.5062c-1.4996,0-2.9916,0-4.4836,0   c-0.0151,1.4996-0.0301,2.9991-0.0377,4.5062c-1.5071,0.0075-3.0067,0.0151-4.5062,0.0302c0,1.4995,0,2.9915,0,4.4836   c1.4995,0.0151,3.0066,0.0302,4.5062,0.0452c0.0151,1.4996,0.0151,2.9991,0.0302,4.4987c1.4996,0,2.9916,0,4.4911,0   c0.0075-1.4996,0.015-2.9991,0.0301-4.5062c1.5071-0.0151,3.0067-0.0226,4.5062-0.0377c0-1.4921,0-2.9916,0-4.4836 C52.0771,25.7792,50.57,25.7792,49.0704,25.7641z"
              />
            </g>
          </svg>
        </a>
      </div>
    </dd>
  </dl>
</div>
`;
